packages/core/content-manager/admin/src/preview/pages/Preview.tsx

```diff
@@ -0,0 +1,197 @@
+import { COLLECTION_TYPES } from '../../constants/collections';
+import { DocumentRBAC } from '../../features/DocumentRBAC';
+import { type UseDocument, useDocument } from '../../hooks/useDocument';
+import { useDocumentLayout } from '../../hooks/useDocumentLayout';
+import { buildValidParams } from '../../utils/api';
+import { useGetPreviewUrlQuery } from '../services/preview';
+
+import type { UID } from '@strapi/types';
+
+/* -------------------------------------------------------------------------------------------------
+ * PreviewProvider
+ * -----------------------------------------------------------------------------------------------*/
+
+interface PreviewContextValue {
+  url: string;
+  mainField: string;
+  document: NonNullable<ReturnType<UseDocument>['document']>;
+  meta: NonNullable<ReturnType<UseDocument>['meta']>;
+  schema: NonNullable<ReturnType<UseDocument>['schema']>;
+}
+
+const [PreviewProvider, usePreviewContext] = createContext<PreviewContextValue>('PreviewPage');
+
+/* -------------------------------------------------------------------------------------------------
+ * PreviewPage
+ * -----------------------------------------------------------------------------------------------*/
+
+const PreviewPage = () => {
+  const { formatMessage } = useIntl();
+
+  // Read all the necessary data from the URL to find the right preview URL
+  const {
+    slug: model,
+    id: documentId,
+    collectionType,
+  } = useParams<{
+    slug: UID.ContentType;
+    id: string;
+    collectionType: string;
+  }>();
+  const [{ query }] = useQueryParams<{
+    plugins?: Record<string, unknown>;
+  }>();
+  const params = React.useMemo(() => buildValidParams(query), [query]);
+
+  if (!collectionType) {
+    throw new Error('Could not find collectionType in url params');
+  }
+
+  if (!model) {
+    throw new Error('Could not find model in url params');
+  }
+
+  // Only collection types must have a documentId
+  if (collectionType === COLLECTION_TYPES && !documentId) {
+    throw new Error('Could not find documentId in url params');
+  }
+
+  const previewUrlResponse = useGetPreviewUrlQuery({
+    params: {
+      contentType: model,
+    },
+    query: {
+      documentId,
+      locale: params.locale,
+      status: params.status as GetPreviewUrl.Request['query']['status'],
+    },
+  });
+
+  const documentResponse = useDocument({
+    model,
+    collectionType,
+    documentId,
+    params,
+  });
+
+  const documentLayoutResponse = useDocumentLayout(model);
+
+  if (
+    documentResponse.isLoading ||
+    previewUrlResponse.isLoading ||
+    documentLayoutResponse.isLoading
+  ) {
+    return <Page.Loading />;
+  }
+
+  if (
+    !previewUrlResponse.data?.data?.url ||
+    previewUrlResponse.error ||
+    documentLayoutResponse.error ||
+    !documentResponse.document ||
+    !documentResponse.meta ||
+    !documentResponse.schema
+  ) {
+    return <Page.NoData />;
+  }
```

Comment:
should we show an error in case there is an issue fetching the previewUrl?

Reply:
oh yes I think I got confused here, I wanted Page.NoData only is no preview URL is provided, and an error screen for the rest. I'll update